cmd/appender/main.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"log"
+	"net/http"
+
+	"github.com/google/go-containerregistry/authn"
+	"github.com/google/go-containerregistry/name"
+	"github.com/google/go-containerregistry/v1"
+	"github.com/google/go-containerregistry/v1/remote"
+	"github.com/google/go-containerregistry/v1/tarball"
+	"github.com/google/go-containerregistry/v1/writable"
+)
+
+var (
+	inputFile  = flag.String("tarball", "", "the tarball to append")
+	srcImage   = flag.String("src-image", "", "The tag of the remote image to append to")
+	dstImage   = flag.String("dst-image", "", "The tag of the resulting remote image")
+	outputFile = flag.String("output-tarball", "", "The output file of the resulting image")
+)
+
+func main() {
+	flag.Parse()
+
+	validateInput()
+
+	src, err := name.ParseReference(*srcImage, name.WeakValidation)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	srcAuth, err := authn.DefaultKeychain.Resolve(src.Context().Registry)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	dst, err := name.ParseReference(*dstImage, name.WeakValidation)
```

Comment:
and see my comment about this being a gray area (probably incorrect) for `tarball.Write`

Reply:
We'll just revert to back to `name.Tag`

> Some (many?) registries don't support push-by-digest anyways.

For my knowledge which ones?

Reply:
GCR does, but IIRC DockerHub doesn't.  Not something I've exhaustively explored, since pushing by digest really requires that folks know what they are doing (and essentially requires custom tooling by a library like this).